### DISCUSSION TITLE
Future option to calculate filled, variant and outline colors with a lighten function

### DISCUSSION THREAD

**Discussion by ReangeloJ** [mantinedev/mantine#805] [Feedback]
Hey thanks for this awesome library <3!

i was wondering if it would be an idea to use a lighten/darken function to determine component colors instead of accessing the index of colors. This in order to change colors more easily, like primaryColor. I believe material ui also does this.

Something like this:
  return {
    border: 'transparent',
    background: theme.fn.themeColor(color, theme.colorScheme === 'dark' ? lighten(primaryColor, 0.8) : lighten(primaryColor, 0.4),
  };
  
  This seems easier than having to change the index positions of your colors as seen in:
  
![image](https://user-images.githubusercontent.com/25207023/152648278-4052e950-044c-4d58-b916-08f5cc8bb2c0.png)

**Comment by rtivital** [comment]
No, this will never happen – components use different colors for different variants.

**Reply by ReangeloJ** [reply]
Okay understood, thanks! :)
